templates/webapps/galaxy/workflow/display.mako

```diff
@@ -104,7 +104,9 @@
                 %if tool:
                   <div class="toolFormTitle">Step ${int(step.order_index)+1}: ${tool.name | h}</div>
                   <div class="toolFormBody">
-                    ${do_inputs( tool.inputs, step.state.inputs, "", step )}
+                    %if hasattr(step, "state"):
```

Comment:
Can you add a comment on when a step has no state attribute?
Is that when a workflow step (or is it a workflow module ?) references a tool parameter that doesn't exist anymore ?
If that's the case, should we render a message that explains this, e.g "This step references an unknown parameter. Please update this step." ?